vizro-core/tests/unit/vizro/models/_components/test_ag_grid.py

```diff
@@ -150,42 +150,75 @@ def test_pre_build_actions_underlying_ag_grid_id(self, ag_grid_with_id, filter_i
 
 class TestBuildAgGrid:
     def test_ag_grid_build_mandatory_only(self, standard_ag_grid, gapminder):
-        ag_grid = vm.AgGrid(id="text_ag_grid", figure=standard_ag_grid)
+        ag_grid = vm.AgGrid(figure=standard_ag_grid)
         ag_grid.pre_build()
         ag_grid = ag_grid.build()
         expected_ag_grid = dcc.Loading(
-            [
-                None,
-                html.Div(
-                    id="text_ag_grid",
-                    children=[html.Div(id="__input_text_ag_grid")],
-                    className="table-container",
-                ),
-            ],
+            html.Div(
+                [
+                    None,
+                    None,
+                    html.Div(
+                        children=[html.Div(id="__input_text_ag_grid")],
```

Comment:
I think we are already aligned on this one already, because that's what I was intending to do, but you just made me notice that I forgot to remove i`d="underlying_ag_grid_id"` in some places :) 

So I've removed the `id` from the unit tests where it was not necessary so `test_XX_grid_build_mandatory_only` and`test_XXXX_build_title_header_footer`. However, there is a separate test called `test_XXX_build_with_underlying_id `in both tables already 👍

Reply:
I like the latest version since it's quite clear and separate what's the `test_XX_grid_build_mandatory_only `, `test_XXXX_build_title_header_footer`, and `test_XXX_build_with_underlying_id`. 👍 

However, I think we should also test (in a separate test) a case when `vm.AgGrid.title` is defined, but the underlying id isn't. In that case, `vm.AgGrid.build` method should return `f"__input_{self.id}"` for the underlying `html.Div`.

Reply:
So, something like this:
```py
   def test_ag_grid_build_with_id_and_without_underlying_id(self, standard_ag_grid, gapminder):
        ag_grid = vm.AgGrid(id="text_ag_grid", figure=standard_ag_grid)
        ag_grid.pre_build()
        ag_grid = ag_grid.build()

        expected_ag_grid = dcc.Loading(
            html.Div(
                [
                    None,
                    None,
                    html.Div(
                        id="text_ag_grid",
                        children=[html.Div(id="__input_text_ag_grid")],
                        className="table-container",
                    ),
                    None,
                ],
                className="figure-container",
            ),
            color="grey",
            parent_className="loading-container",
            overlay_style={"visibility": "visible", "opacity": 0.3},
        )
        assert_component_equal(ag_grid, expected_ag_grid)
```

Reply:
Do you also mind to remove the `filter_interaction_action` from:
 - `test_pre_build_actions_underlying_ag_grid_id` and 
 - `test_ag_grid_build_with_underlying_id` 

in `test_ag_grid` and `test_table` files?

Reply:
Ahh, I see. So the `test_ag_grid_build_mandatory_only` was previously used for this purpose as well. That wasn't clear to me. In this case, I'll add above in and I think having it separate makes it also clearer what that is actually testing!

Reply:
I've parameterized the test :) Will push in a minute